packages/ui/components/form/select/SelectDownshift.tsx

```diff
@@ -0,0 +1,248 @@
+import {
+  useCombobox,
+  UseComboboxGetItemPropsOptions,
+  useMultipleSelection,
+  UseMultipleSelectionGetSelectedItemPropsOptions,
+} from "downshift";
+import React, { ReactElement, ReactNode } from "react";
+
+import { classNames } from "@calcom/lib";
+
+import { Divider } from "../../divider";
+import { TextField } from "../inputs/Input";
+import { Label } from "../inputs/Label";
+
+type StringProperties<T> = {
+  [K in keyof T]: T[K] extends string ? K : never;
+}[keyof T];
+
+interface MultiComboboxProps<T extends Record<string, unknown>> {
+  items: T[];
+  /** The field that will be used to display info */
+  labelKey: keyof T;
+  // gets that can be used to search for items in text box
+  searchableKey?: StringProperties<T>;
+  dividerKey?: keyof T;
+  initalSelectedItems?: T[];
+  onSelectedItemsChange?: (selectedItems: T[]) => void;
+  selectNode?: (
+    item: T,
+    index: number,
+    selection: { isActive: boolean; isKeyboardFocused: boolean },
+    // Downshift doesnt supply types here so we have to say as any
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    getItemProps: (options: UseComboboxGetItemPropsOptions<T>) => any
+  ) => ReactNode;
```

Comment:
Function to pass in your own selection node if you wish to have custom styles 

See storybook file